src/utils.ts

```diff
@@ -219,6 +262,132 @@ export interface Tx {
   scannerUrl: string;
 }
 
+export function txStatusToReceipt(txStatus: TransactionStatus): routes.Receipt {
+  const state = toWormholeTransferState(txStatus.status);
+  const srcChain = toWormholeChainName(txStatus.sourceChain as MayanChainName);
+  const dstChain = toWormholeChainName(txStatus.destChain as MayanChainName);
+
+  const originTxs = txStatus.txs
+    .filter((tx) => {
+      return (
+        // Send from Evm to Solana
+        tx.goals.includes(MayanTransactionGoal.Send) ||
+        // Register for auction on Solana
+        tx.goals.includes(MayanTransactionGoal.Register)
+      );
+    })
+    .map((tx) => {
+      return {
+        chain: srcChain,
+        txid: tx.txHash,
+      };
+    });
+
+  const destinationTxs = txStatus.txs
+    .filter((tx) => {
+      return tx.goals.includes(MayanTransactionGoal.Settle);
+    })
+    .map((tx) => {
+      return {
+        chain: dstChain,
+        txid: tx.txHash,
+      };
+    });
+
+  const vaas: { [key: string]: VAA<"Uint8Array"> } = {};
+  for (const vaaType of possibleVaaTypes) {
+    const key = `${vaaType}SignedVaa`;
+    if (key in txStatus && txStatus[key as keyof TransactionStatus] !== null) {
+      vaas[vaaType] = deserialize(
+        "Uint8Array",
+        encoding.hex.decode(txStatus[key as keyof TransactionStatus])
+      );
+    }
+  }
+
+  switch (state) {
+    case TransferState.SourceInitiated:
+      if ("transfer" in vaas && vaas["transfer"]) {
+        const vaa = vaas["transfer"];
+        const attestation = {
+          id: {
+            emitter: vaa.emitterAddress,
+            sequence: vaa.sequence,
+            chain: vaa.emitterChain,
+          },
+          attestation: vaa,
+        };
+
+        return {
+          from: srcChain,
+          to: dstChain,
+          originTxs,
+          state: TransferState.SourceFinalized,
+          attestation,
+        } satisfies SourceFinalizedTransferReceipt<
+          AttestationReceipt<"WormholeCore">
+        >;
+      }
+      break;
+
+    case TransferState.DestinationInitiated:
+      // VAA to redeem on dest chain
+      if ("redeem" in vaas && vaas["redeem"]) {
```

Comment:
if this check is false, we fall back at the very bottom to returning `SourceInitiated`, when on line 333 we're saying it's past that point and at `DestinationInitiated`. is this correct?

Reply:
its correct in your reading, probably incorrect in behavior in case its somehow not populated